packages/protocol/contracts/L1/libs/LibTaikoToken.sol

```diff
@@ -43,7 +43,7 @@ library LibTaikoToken {
         AddressResolver resolver,
         uint256 amount
     )
-        internal
+        external
```

Comment:
Ideally there's a delay for withdrawing to protect the proposer from a prover that withdraws tokens before the propose, making the propose tx fail.

It is possible to work around this with additional conditions that would stop the propose tx to be valid in PBS, but feels like this should simply be a standard feature so proposers don't have to care about that. A delay of a couple of L1 blocks seems more than enough.

Reply:
- [ ] This has not been done yet, how exactly do you like this to be implemented. I feel like this can be done later.

Reply:
I think something super simple should be okay:
- Prover requests a certain amount of tokens to be withdrawable, this is stored in a mapping and this simply overwrites any previously requested amount. This also sets the timestamp when the amount will be withdrawable
- The withdraw function now simply checks that the amount to withdraw was requested earlier and it's now possible to withdraw that amount.

Agreed that it can be done later, just necessary to make the proposer <-> prover completely trustless.

Reply:
https://github.com/taikoxyz/taiko-mono/issues/15207

Filed a feature request